### PR TITLE
Address self-review for new RPs

### PR DESCRIPTION
**Pull request by ArthurMa1978**

### CODE REVIEW

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs [hunk 1]
```diff
@@ -21,28 +21,28 @@
...
-    /// To get a <see cref="NamespaceNotificationHubAuthorizationRuleCollection" /> instance call the GetNamespaceNotificationHubAuthorizationRules method from an instance of <see cref="NotificationHubResource" />.
+    /// A class representing a collection of <see cref="NotificationHubAuthorizationRuleResource" /> and their operations.
+    /// Each <see cref="NotificationHubAuthorizationRuleResource" /> in the collection will belong to the same instance of <see cref="NotificationHubResource" />.
+    /// To get a <see cref="NotificationHubAuthorizationRuleCollection" /> instance call the GetNotificationHubAuthorizationRules method from an instance of <see cref="NotificationHubResource" />.
     /// </summary>
-    public partial class NamespaceNotificationHubAuthorizationRuleCollection : ArmCollection, IEnumerable<NamespaceNotificationHubAuthorizationRuleResource>, IAsyncEnumerable<NamespaceNotificationHubAuthorizationRuleResource>
+    public partial class NotificationHubAuthorizationRuleCollection : ArmCollection, IEnumerable<NotificationHubAuthorizationRuleResource>, IAsyncEnumerable<NotificationHubAuthorizationRuleResource>
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs:28]: The namespace is `NotificationHubs`, and we used `NotificationHub` here and `NotificationHubs` in some other places like `NotificationHubsSku`. It doesn't keep consistent, maybe we should use `NotificationHubs` as the namespace and `NotificationHub` as the prefix in the resources and models.

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs [hunk 2]
```diff
@@ -65,17 +65,17 @@ internal static void ValidateResourceId(ResourceIdentifier id)
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <exception cref="ArgumentException"> <paramref name="authorizationRuleName"/> is an empty string, and was expected to be non-empty. </exception>
         /// <exception cref="ArgumentNullException"> <paramref name="authorizationRuleName"/> or <paramref name="sharedAccessAuthorizationRuleCreateOrUpdateParameters"/> is null. </exception>
-        public virtual async Task<ArmOperation<NamespaceNotificationHubAuthorizationRuleResource>> CreateOrUpdateAsync(WaitUntil waitUntil, string authorizationRuleName, SharedAccessAuthorizationRuleCreateOrUpdateParameters sharedAccessAuthorizationRuleCreateOrUpdateParameters, CancellationToken cancellationToken = default)
+        public virtual async Task<ArmOperation<NotificationHubAuthorizationRuleResource>> CreateOrUpdateAsync(WaitUntil waitUntil, string authorizationRuleName, SharedAccessAuthorizationRuleCreateOrUpdateParameters sharedAccessAuthorizationRuleCreateOrUpdateParameters, CancellationToken cancellationToken = default)
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs:68]: We should rename the body parameter.

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubData.cs [hunk 3]
```diff
@@ -56,21 +57,21 @@ internal NotificationHubResourceData(ResourceIdentifier id, string name, Resourc
         /// <summary> The NotificationHub name. </summary>
         public string NamePropertiesName { get; set; }
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubData.cs:58]: Bad name, can we rename it?

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs [hunk 4]
```diff
@@ -13,16 +13,16 @@
 
 namespace Azure.ResourceManager.NotificationHubs
 {
-    /// <summary> A class representing the NamespaceResource data model. </summary>
-    public partial class NamespaceResourceData : TrackedResourceData
+    /// <summary> A class representing the NotificationHubNamespace data model. </summary>
+    public partial class NotificationHubNamespaceData : TrackedResourceData
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs:17]: The property name `NamePropertiesName` look weird, I think we can rename it.

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs [hunk 5]
```diff
@@ -78,19 +78,19 @@ internal NamespaceResourceData(ResourceIdentifier id, string name, ResourceType
         /// <summary> The time the namespace was updated. </summary>
         public DateTimeOffset? UpdatedOn { get; set; }
         /// <summary> Endpoint you can use to perform NotificationHub operations. </summary>
-        public string ServiceBusEndpoint { get; set; }
+        public Uri ServiceBusEndpoint { get; set; }
         /// <summary> The Id of the Azure subscription associated with the namespace. </summary>
         public string SubscriptionId { get; set; }
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs:83]: Should we change the type to `Guid`?

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs [hunk 6]
```diff
@@ -11,11 +11,11 @@
 namespace Azure.ResourceManager.NotificationHubs.Models
 {
     /// <summary> Parameters supplied to the Check Name Availability for Namespace and NotificationHubs. </summary>
-    public partial class CheckAvailabilityParameters : TrackedResourceData
+    public partial class NotificationHubNameAvailabilityContent : TrackedResourceData
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs:14]: This model is strange. Usually we will have a name and resource type property in NameAvailabilityContent, why does this model contain sku and IsAvailiable? And it inherits from TrackedResourceData, which I think is not reasonable.

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityResult.cs [hunk 7]
```diff
@@ -12,15 +12,15 @@
...
 {
     /// <summary> Description of a CheckAvailability resource. </summary>
-    public partial class CheckAvailabilityResult : TrackedResourceData
+    public partial class NotificationHubNameAvailabilityResult : TrackedResourceData
     {
-        /// <summary> Initializes a new instance of CheckAvailabilityResult. </summary>
+        /// <summary> Initializes a new instance of NotificationHubNameAvailabilityResult. </summary>
         /// <param name="location"> The location. </param>
-        public CheckAvailabilityResult(AzureLocation location) : base(location)
+        public NotificationHubNameAvailabilityResult(AzureLocation location) : base(location)
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityResult.cs:19]: Similar question as NotificationHubNameAvailabilityContent, why does this model inherit from TrackedResourceData and contain a sku property?

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubPatch.cs [hunk 8]
```diff
@@ -55,21 +56,21 @@ internal NotificationHubResourcePatch(ResourceIdentifier id, string name, Resour
         /// <summary> The NotificationHub name. </summary>
         public string NamePropertiesName { get; set; }
```
Comment (Yao725) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubPatch.cs:57]: Bad name, might need to rename.

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs [hunk 9]
```diff
@@ -1,354 +1,371 @@
...
+    public partial class NotificationHubData : Azure.ResourceManager.Models.TrackedResourceData
+    {
+        public NotificationHubData(Azure.Core.AzureLocation location) : base (default(Azure.Core.AzureLocation)) { }
+        public Azure.ResourceManager.NotificationHubs.Models.NotificationHubAdmCredential AdmCredential { get { throw null; } set { } }
+        public Azure.ResourceManager.NotificationHubs.Models.NotificationHubApnsCredential ApnsCredential { get { throw null; } set { } }
+        public System.Collections.Generic.IList<Azure.ResourceManager.NotificationHubs.Models.SharedAccessAuthorizationRuleProperties> AuthorizationRules { get { throw null; } }
+        public Azure.ResourceManager.NotificationHubs.Models.NotificationHubBaiduCredential BaiduCredential { get { throw null; } set { } }
+        public Azure.ResourceManager.NotificationHubs.Models.NotificationHubGcmCredential GcmCredential { get { throw null; } set { } }
+        public Azure.ResourceManager.NotificationHubs.Models.NotificationHubMpnsCredential MpnsCredential { get { throw null; } set { } }
+        public string NamePropertiesName { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs:80]: Rename or remove this property if it's duplicate with the `name` property from TrackedResourceData.

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs [hunk 10]
```diff
@@ -1,354 +1,371 @@
...
-        public NamespaceResourceData(Azure.Core.AzureLocation location) : base (default(Azure.Core.AzureLocation)) { }
+        public NotificationHubNamespaceData(Azure.Core.AzureLocation location) : base (default(Azure.Core.AzureLocation)) { }
         public System.DateTimeOffset? CreatedOn { get { throw null; } set { } }
-        public bool? Critical { get { throw null; } set { } }
         public string DataCenter { get { throw null; } set { } }
-        public bool? Enabled { get { throw null; } set { } }
+        public bool? IsCritical { get { throw null; } set { } }
+        public bool? IsEnabled { get { throw null; } set { } }
         public string MetricId { get { throw null; } }
         public string NamePropertiesName { get { throw null; } set { } }
```
Comment (fengzhou-msft) [inline sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs:147]: Rename or remove this property.

File: sdk/resourcemover/Azure.ResourceManager.Migrate/api/Azure.ResourceManager.Migrate.netstandard2.0.cs [hunk 11]
```diff
@@ -42,32 +42,32 @@ public partial class MoveCollectionResource : Azure.ResourceManager.ArmResource
...
-        public virtual Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Migrate.Models.OperationStatus> Prepare(Azure.WaitUntil waitUntil, Azure.ResourceManager.Migrate.Models.PrepareContent content = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
-        public virtual System.Threading.Tasks.Task<Azure.ResourceManager.ArmOperation<Azure.ResourceManager.Migrate.Models.OperationStatus>> PrepareAsync(Azure.WaitUntil waitUntil, Azure.ResourceManager.Migrate.Models.PrepareContent content = null, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
+        public virtual Azure.Response<Azure.ResourceManager.Migrate.Models.RequiredForResourcesList> GetRequiredFor(string sourceId, System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken)) { throw null; }
```
Comment (fengzhou-msft) [inline sdk/resourcemover/Azure.ResourceManager.Migrate/api/Azure.ResourceManager.Migrate.netstandard2.0.cs:59]: The method name is a bit confusing as RequiredFor is not a noun after Get.

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs (hunk 10 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs:147]: fixed

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubPatch.cs (hunk 8 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubPatch.cs:57]: fixed

File: sdk/resourcemover/Azure.ResourceManager.Migrate/api/Azure.ResourceManager.Migrate.netstandard2.0.cs (hunk 11 above)
Comment (ArthurMa1978) [reply sdk/resourcemover/Azure.ResourceManager.Migrate/api/Azure.ResourceManager.Migrate.netstandard2.0.cs:59]: Renamed to `GetRequiredForResources`

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs (hunk 5 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs:155]: We keep the SubscriptionId as string for all RPs

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs (hunk 4 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubNamespaceData.cs:17]: fixed

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubData.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubData.cs]: fixed

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs (hunk 2 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs]: change the suffix to `Content`

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs (hunk 9 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/api/Azure.ResourceManager.NotificationHubs.netstandard2.0.cs]: fixed

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs (hunk 1 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/NotificationHubAuthorizationRuleCollection.cs:28]: Good catch, the `NotificationHubs` models are codegen auto-prepend models, fixed

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs]: Good catch, check the swagger I found this is not check the `name` but the resource, so this is why contains `sku` and other properties, and the inherit is caused by codegen's property reference match

Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs]: Remove the `Name` from the model name to `NotificationHubAvailabilityContent`

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityResult.cs (hunk 7 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityResult.cs]: As above, renamed it to `NotificationHubAvailabilityResult`

Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityResult.cs]: Also renamed the corresponding methods

File: sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs (hunk 6 above)
Comment (ArthurMa1978) [reply sdk/notificationhubs/Azure.ResourceManager.NotificationHubs/src/Generated/Models/NotificationHubNameAvailabilityContent.cs]: Also renamed the corresponding methods

**Review by Yao725 — APPROVED** [review]

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**